### DISCUSSION TITLE
Editor crashes when building assets

### DISCUSSION THREAD

**Discussion by cku239** [o3de/o3de#2054] [Help Wanted]
Everything built fine. Tried to create my first project. Just accepted all defaults. Editor crashes whilst building assets. So where do I go from here?

logs show.;-

C:\Users\CKU239\Repos\o3de\Code\Framework\AzCore\AzCore/std/smart_ptr/intrusive_ptr.h(183): (5988) 'class AZ::DX12::CommandQueue &__cdecl AZStd::intrusive_ptr<class AZ::DX12::CommandQueue>::operator *(void) const'
(System) - You can't dereference a null pointer

Exception Description: EXCEPTION_ACCESS_VIOLATION, Attempt to read from address 0x00000176

The memory could not be "read"

Call Stack Trace:
 7) 00007FFD0812C247 (KERNELBASE) : UnhandledExceptionFilter
 6) 00007FFD0A695270 (ntdll) : memset
 5) 00007FFD0A67C776 (ntdll) : _C_specific_handler
 4) 00007FFD0A69217F (ntdll) : _chkstk
 3) 00007FFD0A641454 (ntdll) : RtlRaiseException
 2) 00007FFD0A690CAE (ntdll) : KiUserExceptionDispatcher
 1) C:\Users\CKU239\Repos\o3de\Gems\Atom\RHI\DX12\Code\Source\RHI\CommandQueue.cpp (259) : AZ::DX12::CommandQueue::GetPlatformQueue

**Comment by amzn-tommy** [comment]
This looks like an asset that is critical for Atom to run has not been processed yet. Allow the asset processor to finish running, and that usually resolves the issue. We still have work to do to make sure all critical assets are processed first and make it so the editor can run before the AP has finished processing all the assets.

**Reply by raydentek** [reply]
Hi, I have a similar issue with Editor not running at all after building and registering a project and launching via Editor.exe and project path param. Although no crash, just no sign of it even running / opening. 

- All the other processes seem to be successful, although building a project from the Project Manager also stops abruptly around 20%, then the project folder is around 250MB, while a CLI generated one has 1GB. 
- Launching the project launcher starts Asset Processor which sometimes fails to process some assets, either a Level or some Tif images from a Gem. 
Hopefully a solution can be found, but I'm not in the mood to investigate this at the moment. Perhaps anybody has a clue for me to follow? Thanks!

**Reply by raydentek** [reply]
When launching Editor, nothing happens, then launching Asset Preprocessor through [project_name].GameLauncher.exe gives errors such as the following (Prefabs/Default_Level.prefab). Based on your note, the Editor can't start before the assets are processed. So far I have not managed to get the Editor to launch, and I've been creating some variations of projects. How do you think I can get around this? Thanks.

12.07.2021 00:43 | ERROR | >	D:/RAYDEN/GITHUB/O3DE/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabDomUtils.cpp(152): 'bool __cdecl AzToolsFramework::Prefab::PrefabDomUtils::LoadInstanceFromPrefabDom(class AzToolsFramework::Prefab::Instance &,const class rapidjson_ly::GenericDocument<struct rapidjson_ly::UTF8<char>,class rapidjson_ly::MemoryPoolAllocator<class rapidjson_ly::CrtAllocator>,class rapidjson_ly::CrtAllocator> &,class AZStd::vector<class AZ::Data::Asset<class AZ::Data::AssetData>,class AZStd::allocator> &,enum AzToolsFramework::Prefab::PrefabDomUtils::LoadInstanceFlags)' from AssetBuilder

12.07.2021 00:43 | ERROR | >	Failed to de-serialize Prefab Instance from Prefab DOM. Unable to proceed. from AssetBuilder

12.07.2021 00:43 | ERROR | Prefabs: Trace::Error from AssetBuilder

12.07.2021 00:43 | ERROR | >	D:/RAYDEN/GITHUB/O3DE/o3de/Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/PrefabCatchmentProcessor.cpp(85): 'void __cdecl AzToolsFramework::Prefab::PrefabConversionUtils::PrefabCatchmentProcessor::ProcessPrefab(class AzToolsFramework::Prefab::PrefabConversionUtils::PrefabProcessorContext &,class AZStd::basic_string_view<char,struct AZStd::char_traits<char> >,class rapidjson_ly::GenericDocument<struct rapidjson_ly::UTF8<char>,class rapidjson_ly::MemoryPoolAllocator<class rapidjson_ly::CrtAllocator>,class rapidjson_ly::CrtAllocator> &,enum AZ::DataStream::StreamType)' from AssetBuilder

12.07.2021 00:43 | ERROR | >	Failed to convert prefab 'Default_Level' to a spawnable. from AssetBuilder

12.07.2021 00:43 | ERROR | Prefab Builder: Trace::Error from AssetBuilder

12.07.2021 00:43 | ERROR | >	D:/RAYDEN/GITHUB/O3DE/o3de/Gems/Prefab/PrefabBuilder/PrefabBuilderComponent.cpp(272): 'bool __cdecl AZ::Prefab::PrefabBuilderComponent::ProcessPrefab(const class AZStd::unordered_set<class AZ::Crc32,struct AZStd::hash<class AZ::Crc32>,struct AZStd::equal_to<class AZ::Crc32>,class AZStd::allocator> &,const char *,class AZ::IO::PathView,const struct AZ::Uuid &,class rapidjson_ly::GenericDocument<struct rapidjson_ly::UTF8<char>,class rapidjson_ly::MemoryPoolAllocator<class rapidjson_ly::CrtAllocator>,class rapidjson_ly::CrtAllocator> &,class AZStd::vector<struct AssetBuilderSDK::JobProduct,class AZStd::allocator> &)' from AssetBuilder

12.07.2021 00:43 | ERROR | >	Failed to fully process the target prefab. from AssetBuilder

**Reply by amznestebanpapp** [reply]
Can you try with latest of stabilization/2106? there was a fix going in a couple days ago that fixed some missing dependencies.

**Reply by dsheashea** [reply]
I tried to create a new project using stabilzation/2106 and I am getting the same: ```Exception Description: EXCEPTION_ACCESS_VIOLATION, Attempt to read from address 0x00000120``` .  I also followed along the https://o3de.org/docs/welcome-guide/setup/setup-from-github/ defaults. I did see the editor window open for a split second before crashing which was different than when I tried to create a new project on the development branch.

**Reply by rooswouter** [reply]
> Can you try with latest of stabilization/2106? there was a fix going in a couple days ago that fixed some missing dependencies.

I encountered the same problems as described, and switching to stabilization/2106 fixed the issue for me, all is running correctly now, thank you.
